src/app.js

```diff
@@ -2,0 +1,20 @@
+const fs = require('fs');
+const path = require('path');
+
+let destination = process.argv[3];
+const sourceFile = process.argv[2];
+
+try {
+  const destinationIsDirectory =
+    fs.existsSync(destination) && fs.statSync(destination).isDirectory();
+
+  if (destinationIsDirectory) {
+    const newFileName = path.basename(sourceFile);
+
+    destination = path.join(destination, newFileName);
+  }
+
+  fs.renameSync(sourceFile, destination);
```

Comment:
The `fs.renameSync` function will throw an error if the source file does not exist or if there are permission issues. Ensure that the source file exists before attempting to rename it, or handle such errors gracefully.